From b19f0b09f10984ea9a376275da552f677239ea09 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Tue, 5 Feb 2013 14:22:32 -0600
Subject: [PATCH 187/254] arm: socfpga: enable I2C EEPROM support in dts

Upstream: git://git.rocketboards.org/linux-socfpga.git

Enable 24C32 EEPROM in defconfig and device tree.

Signed-off-by: Alan Tull <atull@altera.com>
(cherry picked from commit 2dd6b8e27bb3ae2c6e1a3a240bac66e1b6d20910)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/socfpga_cyclone5.dts |    6 ++++++
 arch/arm/configs/socfpga_defconfig     |    1 +
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dts b/arch/arm/boot/dts/socfpga_cyclone5.dts
index 4b9fee6..58bd36e 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dts
@@ -141,4 +141,10 @@
 		height = <2>;
 		width = <16>;
 	};
+
+	eeprom@51 {
+		compatible = "atmel,24c32";
+		reg = <0x51>;
+		pagesize = <32>;
+	};
 };
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 5ce9834..ee51f82 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -117,3 +117,4 @@ CONFIG_NEWHAVEN_LCD=y
 CONFIG_I2C_DEBUG_CORE=y
 CONFIG_I2C_DEBUG_ALGO=y
 CONFIG_I2C_DEBUG_BUS=y
+CONFIG_EEPROM_AT24=y
-- 
1.7.5.4

